src/audio/asrc/asrc.c

```diff
@@ -0,0 +1,615 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2019 Intel Corporation. All rights reserved.
+
+#include <sof/audio/asrc/asrc_farrow.h>
+#include <sof/audio/buffer.h>
+#include <sof/audio/component.h>
+#include <sof/audio/format.h>
+#include <sof/audio/pipeline.h>
+#include <sof/debug/panic.h>
+#include <sof/drivers/ipc.h>
+#include <sof/lib/alloc.h>
+#include <sof/lib/cache.h>
+#include <sof/math/numbers.h>
+#include <sof/trace/trace.h>
+#include <sof/common.h>
+#include <sof/list.h>
+#include <sof/platform.h>
+#include <sof/string.h>
+#include <ipc/control.h>
+#include <ipc/stream.h>
+#include <ipc/topology.h>
+#include <user/trace.h>
+#include <errno.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#define trace_asrc(__e, ...)					\
+	trace_event(TRACE_CLASS_ASRC, __e, ##__VA_ARGS__)
+#define tracev_asrc(__e, ...)					\
+	tracev_event(TRACE_CLASS_ASRC, __e, ##__VA_ARGS__)
+#define trace_asrc_error(__e, ...)				\
+	trace_error(TRACE_CLASS_ASRC, __e, ##__VA_ARGS__)
+
+/* asrc component private data */
+struct comp_data {
+	uint32_t sink_rate;
+	uint32_t source_rate;
+	uint32_t sink_format;
+	uint32_t source_format;
+	uint32_t src_size;
```

Comment:
src is not an abbreviation for source here, right?

for someone reading linearly, you have source_format followed by src_size, this can be confusing.

Reply:
I think I'll rename it to asrc_size, the same for src_obj.

Reply:
done!